src/amqp_management.c

```diff
@@ -303,14 +340,54 @@ static void on_message_receiver_state_changed(const void* context, MESSAGE_RECEI
         break;
 
     case MESSAGE_RECEIVER_STATE_OPEN:
-        amqp_management_instance->receiver_connected = 1;
-        (void)send_operation_messages(amqp_management_instance);
+        switch (amqp_management_instance->amqp_management_state)
+        {
+        default:
+        case AMQP_MANAGEMENT_STATE_ERROR:
+            /* do nothing */
+            break;
+
+        case AMQP_MANAGEMENT_STATE_OPENING:
+            amqp_management_instance->receiver_connected = -1;
+
+            if ((amqp_management_instance->sender_connected != 0) &&
+                (amqp_management_instance->receiver_connected != 0))
```

Comment:
Actually only one should be set will, the sender one for message sender and the receiver one for messagereceiver. I'll fix that.